cluster/cluster.go

```diff
@@ -0,0 +1,83 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/hashicorp/raft"
+	"github.com/narvikd/errorskit"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"log"
+	"nubedb/api/proto"
+	"nubedb/cluster/consensus/fsm"
+	"nubedb/internal/config"
+	"time"
+)
+
+func Execute(cfg config.Config, consensus *raft.Raft, payload *fsm.Payload) error {
+	payloadData, errMarshal := json.Marshal(&payload)
+	if errMarshal != nil {
+		return errorskit.Wrap(errMarshal, "couldn't marshal data to send it to the DB cluster")
+	}
+
+	if consensus.State() != raft.Leader {
+		return handleForwardLeaderFuture(cfg, consensus, payload)
+	}
+
+	return applyLeaderFuture(consensus, payloadData)
+}
+
+func applyLeaderFuture(consensus *raft.Raft, payloadData []byte) error {
+	const timeout = 500 * time.Millisecond
```

Comment:
Resolver is 200ms/300ms but this is 500ms

Reply:
Resolver is 200ms because it's a ping/resolve request. This is 500ms because it's a networking operation. I think half a second is a good measure to return a timeout error